src/components/AgregarCliente.vue

```diff
@@ -150,7 +160,51 @@ export default {
     };
   },
   methods: {
-    ...mapActions(["agregarCliente"])
+    ...mapActions(["agregarCliente"]),
+    agregarCliente(){
+      /*
+      Variables
+      */
+      var nombreClientes = document.getElementById("nombreCliente").value;
+      var ciudad = document.getElementById("ciudad").value;
+      var estado = document.getElementById("estado").value;
+      var pais = document.getElementById("pais").value;
+      var correo = document.getElementById("correo").value;
+      var direccion = document.getElementById("direccion").value;
+      var codigoPostal = document.getElementById("codigoPostal").value;
+      var terminoPago = document.getElementById("terminoPago").value;
+      var palabraTraduccion = document.getElementById("palabraTraduccion").value;
+      var palabraEdicion = document.getElementById("palabraEdicion").value;
+      var palabraProofreading = document.getElementById("palabraProofreading").value;
```

Comment:
Todo esto podría simplificarse ampliamente metiendo los nombres de los elementos en un array, no recuerdo la sintaxis Javascript exacta pero sería algo como:
```
var camposAValidar = [
    "nombreCliente",
    "ciudad",
    "estado",
    // etc...
];
```

Y luego iterar ese array preguntando por algún campo vacío:
```
for each (var campo in camposAValidar) {
    camposVacios = camposVacios && campo.value != "";
}
```

Lo mismo aplicado a los numéricos.

Reply:
entiendo lo que decis, pero no estaria entendiendo mucho como esto simplificaria el tema.
ahora me pongo a hacerlo de todas formas

Reply:
Me faltó una cosita en el for each:
```
for each (var campo in camposAValidar) {
    camposVacios = camposVacios && document.getElementById(campo).value != "";
}
```

De todos modos, la simplifcación viene de:
1) No queda un if gigante preguntando individualmente por cada uno de los campos.
2) De hecho, se elimina completamente un bloque if-else.
3) En caso de necesitar validar un nuevo campo, sólo es necesario agregar su id en la lista. Con el esquema anterior hubiera habido que declarar una nueva variable donde almacenarlo y adicionalmente agregarlo al if gigante.

Reply:
ahora los problemas de este metodo
al hacer el for each entro a comparar en booleano los campos vacios para despues poder mostrar el boton
y me queda la siguiente tabla de verdad
falso = falso && verdadero;
esto es 
camposVacios = camposVacios && document.getElementById(campo).value != ""; (una vez que se complete el campo) 
de este modo nunca estoy saliendo del false, ya que tengo inicializada la variable en false. 
entiendo perfectamente lo que estas diciendo pero corri unas pruebas en incognito (sin cache) y no funciono 
las pruebas anteriores que funciono fueron con el cache, y funcionaba todo :/

Reply:
Y entonces Inicializá `camposVacios` en `true` y listo!

Reply:
me gustaria saber como hiciste para que camposvacios este as mas fachero...
cuestion, ya esta.
tambien habia que hacer 2 arrays ya que una palabra/letra no puede ser mayor que 0

Reply:
Jaja qué sé yo...
Sí, obviamente, uno por tipo de dato. Pero al menos así, cuando hay que agregar algo, es en un solo lugar.